shared/js/blobview.js

```diff
@@ -6,6 +6,15 @@ var BlobView = (function() {
     throw Error(msg);
   }
 
+  var decoderCache = {};
+  function getDecoder(encoding) {
+    if (encoding in decoderCache) {
+      return decoderCache[encoding];
+    }
+    var decoder = decoderCache[encoding] = new TextDecoder(encoding);
+    return decoder;
+  }
```

Comment:
There might be a tradeoff between performance and memory usage here. You're going for higher peformance with potentially higher memory usage.  I don't know anything about the TextDecoder implementation, though, so maybe there isn't really any real memory downside to keeping decoder instances around.  On the other hand, I don't know if they are at all expensive to construct new instances. It might be fine to just call new TextDecoder() each time.

I'm fine with doing it this way, but would be finer with it if you had some data (or knowledge of the impl) to justify using this cache.